x-pack/plugins/alerting/server/task_runner/task_runner.test.ts

```diff
@@ -205,6 +208,45 @@ describe('Task Runner', () => {
     expect(call.tags).toEqual(['alert-', '-tags']);
     expect(call.createdBy).toBe('alert-creator');
     expect(call.updatedBy).toBe('alert-updater');
+    expect(call.rule).not.toBe(null);
+    expect(call.rule.name).toBe('alert-name');
+    expect(call.rule.tags).toEqual(['alert-', '-tags']);
+    expect(call.rule.consumer).toBe('bar');
+    expect(call.rule.enabled).toBe(true);
+    expect(call.rule.schedule).toMatchInlineSnapshot(`
+    Object {
+      "interval": "10s",
+    }
+    `);
+    expect(call.rule.createdBy).toBe('alert-creator');
+    expect(call.rule.updatedBy).toBe('alert-updater');
+    expect(call.rule.createdAt).toBe(mockDate);
+    expect(call.rule.updatedAt).toBe(mockDate);
+    expect(call.rule.notifyWhen).toBe('onActiveAlert');
+    expect(call.rule.throttle).toBe(null);
+    expect(call.rule.producer).toBe('alerts');
+    expect(call.rule.ruleTypeId).toBe('test');
+    expect(call.rule.ruleTypeName).toBe('My test alert');
+    expect(call.rule.actions).toMatchInlineSnapshot(`
```

Comment:
ah, interesting!

I seem like I remember some conversations with @mikecote where we were a little concerned about rules knowing a lot about their context.  Some of it just to keep rule executors from having to deal with too much info, but for actions specifically, could we be leaking any info?  In theory action secrets are safe in this context, since they aren't passed in here, just the params.  But it is possible that someone would embed some secrets in something like a webhook header.  Which would then be visible here.

I'm trying to think of good reason to actually pass this in, but failing to come up with one.  Thinking maybe we should remove it based on the security aspect.  @mikecote?

Reply:
My main concern is passing the `actions` enables rule types to change the execution behaviour based on what actions the user has configured. I've been pushing for rule executors to just query and flag things worth alerting on and to let the alerting framework handle the actions (notify every, mapping params, per instance, etc.). This principle facilitates future capabilities like rule simulation, alert summary, alert digest, notify after x amount of times, because of the limited context they know about the rule, forcing things to become `params` instead of `rule` properties where in the future we could call the rule executor without a rule to run a simulation or something.

Reply:
Discussed this with @mikecote offline and we are ok with moving forward on this for now.